provisioning/image_customization.go

```diff
@@ -0,0 +1,164 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+
+	metal3iov1alpha1 "github.com/openshift/cluster-baremetal-operator/api/v1alpha1"
+)
+
+const (
+	ironicBaseUrl    = "IRONIC_BASE_URL"
+	ironicAgentImage = "IRONIC_AGENT_IMAGE"
+)
+
+func setIronicBaseUrl(name string, info *ProvisioningInfo) corev1.EnvVar {
+	config := info.ProvConfig.Spec
+	if config.ProvisioningNetwork != metal3iov1alpha1.ProvisioningNetworkDisabled && !config.VirtualMediaViaExternalNetwork {
+		return corev1.EnvVar{
+			Name:  name,
+			Value: "http://" + config.ProvisioningIP,
+		}
+	} else {
+		hostIP, err := GetPodHostIP(info.Client.CoreV1(), info.Namespace)
+		if err == nil && hostIP != "" {
+			return corev1.EnvVar{
+				Name:  name,
+				Value: "http://" + hostIP,
+			}
+		}
+		return corev1.EnvVar{
+			Name: name,
+		}
+	}
+}
+
+func createImageCustomizationContainer(images *Images, info *ProvisioningInfo) corev1.Container {
+	container := corev1.Container{
+		Name:            "image-customization-controller",
+		Image:           images.ImageCustomizationController,
+		Command:         []string{"/image-customization-controller"},
+		ImagePullPolicy: "IfNotPresent",
+		Env: []corev1.EnvVar{
+			setIronicBaseUrl(ironicBaseUrl, info),
+			{
+				Name:  ironicAgentImage,
+				Value: images.IronicAgent,
+			},
+		},
+		Resources: corev1.ResourceRequirements{
+			Requests: corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("5m"),
+				corev1.ResourceMemory: resource.MustParse("50Mi"),
+			},
+		},
+	}
+	return container
+}
+
+func newImageCustomizationPodTemplateSpec(info *ProvisioningInfo, labels *map[string]string) *corev1.PodTemplateSpec {
+	containers := []corev1.Container{
+		createImageCustomizationContainer(info.Images, info),
+	}
+
+	tolerations := []corev1.Toleration{
+		{
+			Key:      "node-role.kubernetes.io/master",
+			Effect:   corev1.TaintEffectNoSchedule,
+			Operator: corev1.TolerationOpExists,
+		},
+		{
+			Key:      "CriticalAddonsOnly",
+			Operator: corev1.TolerationOpExists,
+		},
+		{
+			Key:               "node.kubernetes.io/not-ready",
+			Effect:            corev1.TaintEffectNoExecute,
+			Operator:          corev1.TolerationOpExists,
+			TolerationSeconds: pointer.Int64Ptr(120),
+		},
+		{
+			Key:               "node.kubernetes.io/unreachable",
+			Effect:            corev1.TaintEffectNoExecute,
+			Operator:          corev1.TolerationOpExists,
+			TolerationSeconds: pointer.Int64Ptr(120),
+		},
+	}
+
+	return &corev1.PodTemplateSpec{
+		ObjectMeta: metav1.ObjectMeta{
+			Annotations: podTemplateAnnotations,
+			Labels:      *labels,
+		},
+		Spec: corev1.PodSpec{
+			Containers:        containers,
+			HostNetwork:       true,
+			DNSPolicy:         corev1.DNSClusterFirstWithHostNet,
+			PriorityClassName: "system-node-critical",
+			NodeSelector:      map[string]string{"node-role.kubernetes.io/master": ""},
+			SecurityContext: &corev1.PodSecurityContext{
+				RunAsNonRoot: pointer.BoolPtr(false),
+			},
```

Comment:
i don't think this deployment needs any special privileges, so you could probably remove the above

Reply:
Ok. Fixed.